zlib/CMakeLists.txt

```diff
@@ -4,6 +4,10 @@ else ()
   set (BUILD_EXT_LIBS_TYPE "STATIC")
 endif ()
 
+if (WIN32 AND NOT MINGW)
+  set (BUILD_EXT_LIBS_TYPE "STATIC")
```

Comment:
The more idomatic way to do this is with `-DBUILD_SHARED_LIBS=OFF` to the cmake command.  I believe this is already an option in the top-level CMakeList.txt so changes to zlib and zstd are not needed.

Lastly, ideally users provide their own copies of zlib and zstd.  They are bundled here only as a last resort since they are several versions behind.

Reply:
You can still want to have SZ as a shared library.

Reply:
Should I get rid of the second commit?

Reply:
That's fair. We can leave the second commit